pkg/test/connector.go

```diff
@@ -88,21 +88,36 @@ func (suite *ConnectorTestSuite) TestAll() {
 					for _, p := range planRes.Msg.GetPartitions() {
 						var cursor []byte
 						var count int
-						for {
-							res, err := c.ListData(ctx, connect.NewRequest(&adiomv1.ListDataRequest{
+						var numLoops = 5
+						for i := 0; i < numLoops; i++ {
+							res1, err := c.ListData(ctx, connect.NewRequest(&adiomv1.ListDataRequest{
 								Partition: p,
 								Type:      t,
 								Cursor:    cursor,
 							}))
 							suite.Assert().NoError(err)
-							count += len(res.Msg.GetData())
-							if res.Msg.GetNextCursor() == nil {
-								break
-							}
-							suite.Assert().NotEqual(cursor, res.Msg.GetNextCursor())
-							cursor = res.Msg.GetNextCursor()
+							count += len(res1.Msg.GetData())
+
+							res2, err := c.ListData(ctx, connect.NewRequest(&adiomv1.ListDataRequest{
+								Partition: p,
+								Type:      t,
+								Cursor:    cursor,
+							}))
+							suite.Assert().NoError(err)
+							suite.Assert().Equal(res1.Msg.GetData(), res2.Msg.GetData(), "Repeated calls with the same cursor should return identical data")
```

Comment:
using `proto.Equal` for protos is safer (the 2 objects might evaluate to be not equal even if they contain the same data)